src/features/metadata-based-view/__tests__/MetadataBasedItemList.test.js

```diff
@@ -86,13 +86,35 @@ describe('features/metadata-based-view/MetadataBasedItemList', () => {
         onItemClick,
     };
 
+    const initialState = {
+        editedColumnIndex: -1,
+        editedRowIndex: -1,
+        hoveredRowIndex: -1,
+        hoveredColumnIndex: -1,
+        valueBeingEdited: null,
+    };
+
     const getWrapper = (props = defaultProps) => mount(<MetadataBasedItemList {...props} />);
 
     beforeEach(() => {
         wrapper = getWrapper();
         instance = wrapper.instance();
     });
 
+    describe('componentDidUpdate()', () => {
+        test('should call setState when component gets updated with different props', () => {
```

Comment:
Should we add another test to ensure that `setState` is not called when the props haven't changed, i.e., `isEqual(currentItems, prevItems) === true`?

Reply:
Done